rust_snuba/src/strategies/clickhouse/batch.rs

```diff
@@ -233,7 +237,10 @@ impl HttpBatch {
         // finish stream
         drop(self.sender.take());
         if let Some(handle) = self.result_handle.take() {
-            handle.await??;
+            // timeout on writing a batch to clickhouse after 2 minutes
+            if let Err(_) = timeout(Duration::from_millis(120000), handle).await {
```

Comment:
pls handle the inner errors

```suggestion
            match timeout(Duration::from_millis(120000), handle).await {
                Ok(res) => res??,
                Err(_) => {
```